### ISSUE TITLE
False positive E1121: Too many positional arguments error for Subclassed Enums instanciated with functional API

### ISSUE THREAD

**Issue by fgoudreault** [pylint-dev/pylint#10609]
### Bug description

The following causes the `E1121: Too many positional arguments for constructor call (too-many-function-args)` even though this is valid python code.

```python

from enum import Enum

class CustomEnum(Enum):
    pass

MyEnum = CustomEnum('MyEnum', {'RED': 1})  # E1121
MyEnum2 = Enum('MyEnum2', {'BLUE': 2})     # No errors

class MyEnum3(CustomEnum):    # No errors
    GREEN = 3
```

### Configuration

```ini
Using default config in pristine `venv` with only a `pip install pylint`.
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:12:10: E1121: Too many positional arguments for constructor call (too-many-function-args)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 3.3.8
astroid 3.3.11
Python 3.13.7 (tags/v3.13.7:bcee1c32211, Aug 18 2025, 10:05:53) [GCC 13.3.0]
```

### OS / Environment

Ubuntu 24.04
python 3.13.7 

### Additional dependencies

```python
None
```
